cirq-core/cirq/optimizers/two_qubit_state_preparation.py

```diff
@@ -0,0 +1,114 @@
+# Copyright 2021 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Utility methods for efficiently preparing two qubit states."""
+
+from typing import Sequence, TYPE_CHECKING
+import numpy as np
+
+from cirq import ops, qis
+from cirq.optimizers import decompositions
+
+if TYPE_CHECKING:
+    import cirq
+
+
+def _1q_matrices_to_ops(g0, g1, q0, q1, include_identity=False):
+    ret = []
+    for g, q in zip(map(decompositions.single_qubit_matrix_to_phxz, [g0, g1]), [q0, q1]):
+        if g is not None:
+            ret.append(g.on(q))
+        elif include_identity:
+            ret.append(ops.I.on(q))
+    return ret
+
+
+def prepare_two_qubit_state_using_sqrt_iswap(
+    q0: 'cirq.Qid',
+    q1: 'cirq.Qid',
+    state: 'cirq.STATE_VECTOR_LIKE',
+    *,
+    use_sqrt_iswap_inv: bool = True,
```

Comment:
Nit: iswap_inv instead ?

Reply:
I'd prefer to keep it `use_sqrt_iswap_inv` since it's also consistent with the keyword argument in `two_qubit_matrix_to_sqrt_iswap_operations`